### PR TITLE
Add `use_slot_uris` argument which you can use in gen-doc jinja templates

### PR DESCRIPTION
**Pull request by sujaypatil96**
This PR seeks to resolve issue #791.

The way in which you can enforce the usage of term_ids in URLs is by setting `term_id=True`.

### CODE REVIEW

**Review by cmungall — CHANGES_REQUESTED** [review]
Logic is good, but "term" isn't a term in linkml, instead call the parameter `use_slot_uris`

**Comment by cmungall** [comment]
This also leaves open to use_class_uris in future (not a MIxS reqt yet?)

**Comment by sujaypatil96** [comment]
Hm, yes, that's true. It does let us extrapolate the same functionality to be applicable to classes by exposing a use_class_uris argument. But it isn't a MIxS requirement just yet.

File: linkml/generators/docgen.py [hunk 1]
```diff
@@ -121,7 +124,7 @@ def serialize(self, directory: str = None) -> None:
         for sn, s in sv.all_slots().items():
             if self._is_external(s):
                 continue
-            n = self.name(s)
+            n = self.name(s, use_slot_uris=self.use_slot_uris)
```
Comment (cmungall) [inline linkml/generators/docgen.py:127]: the logic here is odd. use_slot_uris is already an instance attribute on the Generator

**Review by cmungall — APPROVED** [review]

**Pull request merged**